aptos-move/framework/aptos-framework/sources/multisig_account.move

```diff
@@ -0,0 +1,1348 @@
+        vote_transanction(owner, multisig_account, transaction_id, true);
+    }
+
+    /// Reject a multisig transaction.
+    public entry fun reject_transaction(
+        owner: &signer, multisig_account: address, transaction_id: u64) acquires MultisigAccount {
+        vote_transanction(owner, multisig_account, transaction_id, false);
+    }
+
+    /// Generic function that can be used to either approve or reject a multisig transaction
+    public entry fun vote_transanction(
+        owner: &signer, multisig_account: address, transaction_id: u64, approved: bool) acquires MultisigAccount {
+        assert_multisig_account_exists(multisig_account);
+        let multisig_account_resource = borrow_global_mut<MultisigAccount>(multisig_account);
+        assert_is_owner(owner, multisig_account_resource);
+
+        assert!(
+            table::contains(&multisig_account_resource.transactions, transaction_id),
+            error::not_found(ETRANSACTION_NOT_FOUND),
+        );
+        let transaction = table::borrow_mut(&mut multisig_account_resource.transactions, transaction_id);
+        let owner_addr = address_of(owner);
+        if (approved) {
+            assert!(
+                !simple_map::contains_key(&transaction.approvals, &owner_addr),
+                error::already_exists(TRANSACTION_HAS_ALREADY_BEEN_APPROVED)
+            );
+            simple_map::add(&mut transaction.approvals, owner_addr, true);
+            // Revoke rejection.
+            if (simple_map::contains_key(&mut transaction.rejections, &owner_addr)) {
+                simple_map::remove(&mut transaction.rejections, &owner_addr);
+            };
+            emit_event(
+                &mut multisig_account_resource.approve_transaction_events,
+                ApproveTransactionEvent {
+                    owner: owner_addr,
+                    transaction_id,
+                    num_approvals: simple_map::length(&transaction.approvals),
+                }
+            );
+        } else {
+            assert!(
+                !simple_map::contains_key(&transaction.rejections, &owner_addr),
+                error::already_exists(TRANSACTION_HAS_ALREADY_BEEN_REJECTED)
+            );
+            simple_map::add(&mut transaction.rejections, owner_addr, true);
+            // Revoke approval.
+            if (simple_map::contains_key(&mut transaction.approvals, &owner_addr)) {
+                simple_map::remove(&mut transaction.approvals, &owner_addr);
+            };
+            emit_event(
+                &mut multisig_account_resource.reject_transaction_events,
+                RejectTransactionEvent {
+                    owner: owner_addr,
+                    transaction_id,
+                    num_rejections: simple_map::length(&transaction.rejections),
+                }
+            );
+        };
+    }
+
+    /// Remove the next transaction if it has sufficient owner rejections.
+    public entry fun remove_transaction(
```

Comment:
Suggest rename to `execute_rejected_transaction`. Name `remove_transaction` is a little bit confusing since it sounds like removing a transaction with a specific tx id, and the tx id can be reused after execution.